astroquery/cds/core.py

```diff
@@ -0,0 +1,201 @@
+        :param output_format: The output format we want the matching datasets (full record, just IDs, mocpy object)
+        :param get_query_payload: specify if we only want the cds service to return the payload
+        :param cache: boolean
+        :return: The HTTP response returned from the service. All async methods should return the raw HTTP response.
+        """
+        request_payload = dict()
+        if not isinstance(constraints, Constraints):
+            print("Invalid constraints. Must be of MOCServerConstraints type")
+            raise TypeError
+
+        if not isinstance(output_format, OutputFormat):
+            print("Invalid response format. Must be of MOCServerResponseFormat type")
+            raise TypeError
+
+        request_payload.update(constraints.payload)
+        request_payload.update(output_format.request_payload)
+
+        if get_query_payload:
+            return request_payload
+
+        payload_d = {
+            'method': 'GET',
+            'url': self.URL,
+            'timeout': self.TIMEOUT
+        }
+
+        if 'moc' not in request_payload:
+            payload_d.update({'params': request_payload,
+                              'cache': cache})
+            return self._request(**payload_d)
+
+        filename = request_payload['moc']
+        with open(filename, 'rb') as f:
+            request_payload.pop('moc')
+
+            payload_d.update({'params': request_payload,
+                              'cache': False,
+                              'files': {'moc': f}})
+
+            return self._request(**payload_d)
+
+    @staticmethod
+    def _parse_to_float(value):
+        try:
+            return float(value)
+        except (ValueError, TypeError):
+            return value
+
+    @staticmethod
+    def orderipix2uniq(n_order, n_pix):
+        return ((4**n_order) << 2) + n_pix
+
+    @staticmethod
+    def create_mocpy_object_from_json(json_moc):
+        # TODO : just call the MOC.from_json classmethod to get the corresponding mocpy object
+        uniq_interval = IntervalSet()
+        for n_order, n_pix_l in json_moc.items():
+            n_order = int(n_order)
+
+            for n_pix in n_pix_l:
+                uniq_interval.add(__class__.orderipix2uniq(n_order, n_pix))
+
+        moc = MOC.from_uniq_interval_set(uniq_interval)
+        return moc
+
+    @staticmethod
+    def _parse_result_region(response, output_format, verbose=False):
+        """
+        Parse the cds mocserver HTTP response to a more convenient format for python users
+
+        :param response: the HTTP response
+        :param output_format: the output format that the user has specified
+        :param verbose: boolean. if verbose is False then suppress any VOTable related warnings
+        :return: the final parsed response that will be given to the user
+        """
+        if not verbose:
+            commons.suppress_vo_warnings()
+
+        r = response.json()
```

Comment:
generally we prefer to avoid single-character variables.  Perhaps use `result` here instead?  or `json_data`?